kernel/src/main/java/org/kframework/parser/concrete2kore/kernel/Grammar.java

```diff
@@ -549,50 +549,34 @@ public boolean isNullable() {
     public static class RegExState extends PrimitiveState {
         /** The java regular expression pattern. */
         public final RunAutomaton pattern;
+        public final RunAutomaton precedePattern;
         public final RunAutomaton followPattern;
 
         /** The set of terminals (keywords) that shouldn't be parsed as this regular expression. */
 
         public RegExState(String name, NonTerminal nt, Automaton pattern) {
-            this(name, nt, pattern, BasicAutomata.makeEmpty());
+            this(name, nt, BasicAutomata.makeEmpty(), pattern, BasicAutomata.makeEmpty());
         }
 
-        public RegExState(String name, NonTerminal nt, Automaton pattern, Automaton followPattern) {
+        public RegExState(String name, NonTerminal nt, Automaton precedePattern, Automaton pattern, Automaton followPattern) {
             super(name, nt);
             assert pattern != null;
+            this.precedePattern = new RunAutomaton(precedePattern != null ? precedePattern : BasicAutomata.makeEmpty(), false);
             this.pattern = new RunAutomaton(pattern, false);
-            this.followPattern = new RunAutomaton(followPattern, false);
+            this.followPattern = new RunAutomaton(followPattern != null ? followPattern : BasicAutomata.makeEmpty(), false);
         }
 
         // Position is an 'int' offset into the text because CharSequence uses 'int'
-        Set<MatchResult> matches(CharSequence text, int startPosition) {
-            int i = lookingAt(text, startPosition, pattern);
-            if (i >= 0) {
-                int j = lookingAt(text, i, followPattern);
-                if (j < 0) {
-                    return Collections.singleton(new MatchResult(i));
-                }
-            }
-            return Collections.emptySet();
-        }
-
-        int lookingAt(CharSequence text, int startPosition, RunAutomaton pattern) {
-            int state = pattern.getInitialState();
-            int len = text.length();
-
-            int acceptIdx = pattern.isAccept(state) ? startPosition : -1;
-            for(int i = startPosition; i < len; ++i) {
-                int nextState = pattern.step(state, text.charAt(i));
-                if(nextState == -1) {
-                    break;
-                }
-                if (pattern.isAccept(nextState)) {
-                    acceptIdx = i + 1;
-                }
-                state = nextState;
-            }
-
-            return acceptIdx;
+        Set<MatchResult> matches(String text, String reverseText, int startPosition) {
+            int matchedLenght = pattern.run(text, startPosition);
```

Comment:
Also, I'm pretty sure this isn't going to work. The run method matches the entire string, which is why I created tho lookingAt method. Please test this on the C semantics before you assume that this will work correctly.